src/index.ts

```diff
@@ -1,23 +1,25 @@
-let numero: number = 0;
+let cantidadPersonas: number = 270;
+let personas: number[] = new Array(cantidadPersonas);
 let menorValor: number = 18;
 let mayorValor: number = 40;
 let cantMenor: number = 0;
 let cantMayor: number = 0;
-let contador: number = 0;
-let persona: number;
 
-function cargarEdades(menorValor: number, mayorValor: number) {
-  numero = Math.floor(Math.random() * 100);
-  if (menorValor <= numero && numero <= mayorValor) {
-    persona = numero;
+function cargarEdades() {
+  for (let indice = 0; indice < cantidadPersonas; indice++) {
+    let edad =
+      Math.floor(Math.random() * (mayorValor - menorValor + 1)) + menorValor;
+    personas[indice] = edad;
   }
-  return persona;
 }
 
-for (contador = 0; contador < 270; contador++) {
-  if (cargarEdades(menorValor, mayorValor) < 21) {
+cargarEdades();
+for (let indice = 0; indice < cantidadPersonas; indice++) {
+  if (personas[indice] < 21) {
     cantMenor += 1;
-  } else cantMayor += 1;
+  } else {
+    cantMayor += 1;
+  }
 }
```

Comment:
Aca llamamos la funcion **cargarEdades()** para que se ejecute el for y llene el array de personas.
Despues sigue el otro for que calcula la cantidad de personas por edad. Este for recorre todo el array de personas y suma segun sea mayor o menor a 21.